org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/JavaFormatter.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.formatting2.internal;
+
+import org.eclipse.xtext.formatting2.AbstractJavaFormatter;
+import org.eclipse.xtext.formatting2.IFormattableDocument;
+import org.eclipse.xtext.formatting2.internal.formattertestlanguage.IDList;
+import org.eclipse.xtext.formatting2.internal.services.FormatterTestLanguageGrammarAccess;
+
+import com.google.inject.Inject;
+
+/**
+ * @author Arne Deutsch - Initial contribution and API
+ */
+public class JavaFormatter extends AbstractJavaFormatter {
```

Comment:
I don't like the class name. Is that what is generated?

Reply:
Do you have an issue with AbstractJavaFormatter or JavaFormatter or both? The latter is a test class, hence I don't pay so much attention. For the former I'm open for better names, but am not sure for myself ...

Reply:
FormatterTestLanguageJavaFormatter?